src/main/scala/fr/acinq/bitcoin/Crypto.scala

```diff
@@ -251,48 +239,33 @@ object Crypto {
       key.length match {
         case 65 if key(0) == 4 || key(0) == 6 || key(0) == 7 =>
           key(0) = if ((key(64) & 0x01) != 0) 0x03.toByte else 0x02.toByte
-          new PublicKey(ByteVector(key, 0, 33))
+          new PublicKey(Point(ByteVector(key, 0, 33)), true)
         case 33 if key(0) == 2 || key(0) == 3 =>
-          new PublicKey(ByteVector(key, 0, 33))
+          new PublicKey(Point(ByteVector(key, 0, 33)), true)
         case _ =>
           throw new IllegalArgumentException(s"key must be 33 or 65 bytes")
       }
     }
-
-    def apply(point: Point) = new PublicKey(point.toBin(true))
-
-    def apply(point: Point, compressed: Boolean) = new PublicKey(point.toBin(compressed = compressed))
   }
 
+
   /**
     *
-    * @param raw        serialized value of this public key (a point)
+    * @param value EC point
+    * @param compressed if true, key will be serialized in compressed DER format
     */
-  class PublicKey(val raw: ByteVector) extends Serializable {
-    // we always make this very basic check
-    require(isPubKeyValid(raw))
-
-    lazy val compressed = isPubKeyCompressed(raw)
-
-    lazy val value: Point = Point(raw)
+  case class PublicKey(value: Point, compressed: Boolean) extends Serializable {
```

Comment:
I don't think we should make the `compressed` flag an attribute of the case class, because it would be used for comparing public keys, which it should not?

Reply:
Also that's not consistent with the choice you made for `ECPoint`?

Reply:
It's different because a point is just a point, how we encode it has no specific meaning. But in Bitcoin compressed and uncompressed public keys give different hashes and different addresses, so I would say they're different. Uncompressed keys are almost never used now but I thought that defining different types for uncompressed and compressed public keys would be too much of a change.